ADVANCED.md

```diff
@@ -653,6 +654,55 @@ We have more discussion and examples [in our issue here](https://github.com/sw-y
 
 </details>
 
+# Section 4: @types/react and @types/react-dom APIs
+
+## `@types/react`
+
+[Link to `.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts)
+
+**Namespace: React**
+
+Most Commonly Used Interfaces and Types
+
+- `Ref`
+- `ReactElement`
+- `ReactSVGElement`
+- `ReactPortal`
+
+Not Commonly Used but Good to know
+
+- `DetailedReactHTMLElement`
+- `ReactHTMLElement`
+- `DOMElement`
+- `ClassicElement`
+- `ComponentElement`
+- `FunctionComponentElement`
+- `ClassAttributes`
+- `RefAttributes`
+- `Attributes`
+- `RefObject`
+
+Don't use/Internal/Deprecated
+
+- `SFCElement`
+- `SFC`
+- `ComponentState`
+- `LegacyRef`
+- `ReactType`
```

Comment:
Do you have a source for this? Looking the .d.ts source it seems like they explicitly annotate deprecated types and this is not the case for `ReactType`.

 `ReactType` is pretty useful to cover most types that can be passed to `createElement` e.g.
```tsx
function PassThrough(props: { as: ReactType<any> }) {
   const { as: Component } = props;

   return <Component />
}

Reply:
nope it was just a quick first cut based on my experience. thats why i open up this as a PR so we can discuss :)